kedro/extras/datasets/api/api_dataset.py

```diff
@@ -21,83 +21,81 @@ class APIDataSet(AbstractDataSet):
         >>>
         >>> data_set = APIDataSet(
         >>>     url="https://quickstats.nass.usda.gov",
-        >>>     params={
-        >>>         "key": "SOME_TOKEN",
-        >>>         "format": "JSON",
-        >>>         "commodity_desc": "CORN",
-        >>>         "statisticcat_des": "YIELD",
-        >>>         "agg_level_desc": "STATE",
-        >>>         "year": 2000
-        >>>     }
+        >>>     load_args={
+        >>>         "params": {
+        >>>             "key": "SOME_TOKEN",
+        >>>             "format": "JSON",
+        >>>             "commodity_desc": "CORN",
+        >>>             "statisticcat_des": "YIELD",
+        >>>             "agg_level_desc": "STATE",
+        >>>             "year": 2000
+        >>>         }
+        >>>     },
+        >>>     credentials=("username", "password")
         >>> )
         >>> data = data_set.load()
     """
 
-    # pylint: disable=too-many-arguments
     def __init__(
         self,
         url: str,
         method: str = "GET",
-        data: Any = None,
-        params: Dict[str, Any] = None,
-        headers: Dict[str, Any] = None,
-        auth: Union[Iterable[str], AuthBase] = None,
-        json: Union[List, Dict[str, Any]] = None,
-        timeout: int = 60,
-        credentials: Union[Iterable[str], AuthBase] = None,
+        load_args: Dict[str, Any] = None,
+        credentials: Union[Tuple[str, str], List[str]] = None,
```

Comment:
I'm still doubting about this, since technically it works. However, I think it might be confusing to the reader, since `APIDataSet` are automatically constructed from the catalog, and `AuthBases` cannot be inferred this way. So I am wondering, in what cases would this constructor type be used when using Kedro? At this moment, for a user that wants to do a request using an `AuthBase` instance, should imo take a different approach, such as extending `APIDataSet`. However, if you have other ideas, I'll put `AuthBase` in the typing.

Reply:
Very slow reply I know, sorry!! 😅 The use case here would be that the `APIDataSet` is constructed directly using the Python API rather than from the data catalog. In this case you can easily pass arbitrary Python objects to the constructor. This is unusual but does occur and is something we support. So I've put `AuthBase` back in the type hint 🙁